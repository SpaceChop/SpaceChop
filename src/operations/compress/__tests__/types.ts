import CompressConfig from '../types';

describe('should validate compress config', () => {
  describe('should throw on invalid config', () => {
    it('require quality as non decimal number', () => {
      const config = {
        quality: 0.9,
        lossy: true,
      };
      expect(() => CompressConfig.check(config)).toThrowErrorMatchingSnapshot();
    });

    it('require quality between 0-100', () => {
      const config = {
        quality: -1,
        lossy: true,
      };
      expect(() => CompressConfig.check(config)).toThrowErrorMatchingSnapshot();
    });
  });

  it('should now throw on valid config', () => {
    const config = {
      quality: 90,
      lossy: true,
    };
    expect(() => CompressConfig.check(config)).not.toThrow();
  });
});
