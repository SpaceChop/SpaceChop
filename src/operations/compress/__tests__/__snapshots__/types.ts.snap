// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should validate compress config should throw on invalid config require quality as non decimal number 1`] = `"quality must be non decimal number"`;

exports[`should validate compress config should throw on invalid config require quality between 0-100 1`] = `"quality must be between 0 and 100"`;
